flutter_frontend/lib/View/WaitingRoom.dart

```diff
@@ -52,69 +52,135 @@ class _WaitingRoomState extends State<WaitingRoom> {
     );
   }
 
-  Widget _getPlayerIcon(Player player, double width) {
-    Colour colour = player != null ? player.colour : Colour.lightBlue; // TODO: change this to a placeholder
-    String name = player != null ? player.name : '---'; // TODO: change this to a placeholder
-    bool isPlayerReady = player != null && player.isReady; // TODO: change this to a placeholder
-    print("========");
-    print("colour is " + colour.toString());
-    String source = Commons.colourToAssetString(colour);
-
-    return Column(
-      children: [
-        Image(
-          image: AssetImage(source),
-          width: width,
-        ),
-        Text(
-          name,
-          style: TextStyle(
-            color: isPlayerReady ? Colors.red : Colors.grey,
-            fontFamily: "WalterTurncoat",
-            fontSize: smallFontSize,
-          ),
-        )
-      ],
-    );
-  }
-
-  // returns null if index out of range
-  Player _getPlayer(int index) {
-    print("players length " + players.length.toString());
-    if (index >= players.length) {
-      return null;
-    } else {
-      return players[index];
+  Widget _getIcon(Colour colour, String name, double width, double opacity) {
+      String caption = name == null ? 'Waiting...' : name;
```

Comment:
Should we pass 'Waiting...' as the name instead of `null`? Since we set the opacity for the case when the colour is not chosen yet anyway